radixdlt/babylon-wallet-android
# [ABW-3602] Lock screen

## Description
Lock screen is implemented as separate activity. Since we have now 2 activities, app background/foreground detection should happen in application class now. We add `privacyOverlay` to the window when app is just about to go
 to background, and we remove it right after we resume MainActivity (and start lock activity if enabled). Additionally in WalletApp we also display exact same screen content if app is still locked. Each place, privacy overlay, wallet app `FullScreen` composable and lock screen activity use exactly same content. This all is done to avoid app content being shown when app goes to foreground - before lock activity is opened. Still, there are minor issues some time, for example when dialog is open in the app and we background it. 
This PR also adds code to sync enableEdgeToEdge setup with selected theme, similarly as now in android app. It seem that this way app works better when switching theme while app is in the foreground. Thanks @giannis-rdx for the hint!
I left commented code and some unused pieces of code, just in case we will like to explore different implementations, for example one screen, using composable destination as a lock screen. After this PR will get tested and approved, I will clean up stuff that is unused.
About webRTC requests coming in when lock screen is on:
- From my testing it looks that when we do cold start of the app and it is locked immediately, requests are not received until MainActivity goes into display again
- if we lock app by pressing home and then re-open, requests are received by MainActivity and 1st request is displayed below the lock screen 

## How to test

1. Enable/disable lock screen in "preferences"
2. Verify locking/unlocking process on various app screens - normal screens, dialog, in transaction signing
